File: intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.go
func intersect(nums1 []int, nums2 []int) []int {
	sort.Ints(nums1)
	sort.Ints(nums2)

	var (
		i,j int
		ret = []int{}
	)

	for i, j = 0, 0; i < len(nums1) && j < len(nums2); {
		if nums1[i] == nums2[j] {
			ret = append(ret, nums1[i])
			i++
			j++
        } else if nums1[i] < nums2[j] {
            i++
        } else{
            j++
        }
	}
    
	return ret
}

File: intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        c = Counter(nums1)
        output = []
        for n in nums2:
            if c[n] > 0:
                output.append(n)
                c[n] -= 1
        return output
            